flake.nix

```diff
@@ -0,0 +1,253 @@
+                  ./patches/sigsegv-stack.patch
+                  ./patches/macos-unexec.patch
+                ];
+                inherit (self.darwin) sigtool;
+              };
+
+              emacs-26-2 = super.callPackage ./emacs.nix {
+                inherit (source_for "26.2") name src version;
+                inherit latestPackageKeyring;
+                patches = [
+                  ./patches/gnutls-e_again.patch
+                  ./patches/sigsegv-stack.patch
+                  ./patches/macos-unexec.patch
+                ];
+                inherit (self.darwin) sigtool;
+              };
+
+              emacs-26-3 = super.callPackage ./emacs.nix {
+                inherit (source_for "26.3") name src version;
+                inherit latestPackageKeyring;
+                patches = [
+                  ./patches/sigsegv-stack.patch
+                  ./patches/macos-unexec.patch
+                ];
+                inherit (self.darwin) sigtool;
+              };
+
+              emacs-27-1 = super.callPackage ./emacs.nix {
+                inherit (source_for "27.1") name src version;
+                inherit latestPackageKeyring;
+                patches = [
+                  ./patches/sigsegv-stack.patch
+                ];
+                inherit (self.darwin) sigtool;
+              };
+
+              emacs-27-2 = super.callPackage ./emacs.nix {
+                inherit (source_for "27.2") name src version;
+                inherit latestPackageKeyring;
+                patches = [
+                  ./patches/sigsegv-stack.patch
+                ];
+                inherit (self.darwin) sigtool;
+              };
+
+              emacs-28-1 = super.callPackage ./emacs.nix {
+                inherit (source_for "28.1") name src version;
+                inherit latestPackageKeyring;
+                inherit (self.darwin) sigtool;
+              };
+
+              emacs-28-2 = super.callPackage ./emacs.nix {
+                inherit (source_for "28.2") name src version;
+                inherit latestPackageKeyring;
+                inherit (self.darwin) sigtool;
+              };
+
+              emacs-release-snapshot = super.callPackage ./emacs.nix {
+                inherit (source_for "release-snapshot") name src;
+                inherit latestPackageKeyring;
+                version = "29.0.60";
+                srcRepo = true;
+                treeSitter = true;
+                inherit (self.darwin) sigtool;
+              };
+
+              emacs-snapshot = super.callPackage ./emacs.nix {
+                inherit (source_for "snapshot") name src;
+                inherit latestPackageKeyring;
+                version = "30.0.50";
+                srcRepo = true;
+                treeSitter = true;
+                inherit (self.darwin) sigtool;
+              };
+            };
+          in
+          versions // {
+            emacs-ci-versions = builtins.attrNames versions;
```

Comment:
One could use the following command to retrieve the list of Emacs versions available for the platform, so you can omit this version list once you switch to a flake:

```sh
nix eval nix-emacs-ci#packages.x86_64-linux --apply builtins.attrNames
```